src/bin/neptune-cli.rs

```diff
@@ -1,24 +1,47 @@
-use neptune_core::models::blockchain::type_scripts::neptune_coins::NeptuneCoins;
-use neptune_core::models::state::wallet::coin_with_possible_timelock::CoinWithPossibleTimeLock;
-
 use anyhow::{bail, Result};
 use clap::{CommandFactory, Parser};
 use clap_complete::{generate, Shell};
-
 use neptune_core::config_models::data_directory::DataDirectory;
 use neptune_core::config_models::network::Network;
-use neptune_core::models::state::wallet::address::generation_address;
+use neptune_core::models::blockchain::block::block_selector::BlockSelector;
+use neptune_core::models::blockchain::type_scripts::neptune_coins::NeptuneCoins;
+use neptune_core::models::state::wallet::address::ReceivingAddressType;
+use neptune_core::models::state::wallet::coin_with_possible_timelock::CoinWithPossibleTimeLock;
+use neptune_core::models::state::wallet::wallet_status::WalletStatus;
 use neptune_core::models::state::wallet::WalletSecret;
+use neptune_core::rpc_server::RPCClient;
 use std::io;
+use std::io::stdout;
 use std::io::Write;
 use std::net::IpAddr;
 use std::net::SocketAddr;
+use std::str::FromStr;
 use tarpc::{client, context, tokio_serde::formats::Json};
 
-use neptune_core::models::blockchain::block::block_selector::BlockSelector;
-use neptune_core::models::state::wallet::wallet_status::WalletStatus;
-use neptune_core::rpc_server::RPCClient;
-use std::io::stdout;
+// for parsing SendToMany <output> arguments.
+#[derive(Debug, Clone)]
+struct TransactionOutput {
+    address: String,
+    amount: NeptuneCoins,
+}
+
+impl FromStr for TransactionOutput {
+    type Err = anyhow::Error;
+
+    // parses address:amount into TransactionOutput{address, amount}
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts = s.split(':').collect::<Vec<_>>();
+
+        if parts.len() != 2 {
+            anyhow::bail!("Invalid transaction output.  missing :")
+        }
+
+        Ok(Self {
+            address: parts[0].to_string(),
+            amount: NeptuneCoins::from_str(parts[1])?,
+        })
+    }
+}
```

Comment:
`TransactionOutput` here is a private struct within the executable neptune-cli.rs, so it is not part of the neptune-core crate, per se and is not part of a public API.    neptune-cli is presently quite minimal; there aren't any tests in the module at all yet.   That's not to say we shouldn't eventually test it rigorously or that we shouldn't add tests as new features are added, only that it hasn't been the methodology here to date.

I'm unaware how to parse a colon delimited string into a struct with Serde except by implementing Deserialize, which would then do what FromStr is already doing.  I can switch to serde if desired.  FromStr is just a little quicker/easier and got the job done here.

When using FromStr, the reciprocal serialization would be to impl Display, which tnen provides to_string().  I did not do this because there is presently no need.   Again we are in an executable context of neptune-cli, not a crate providing a public API.  Since we do not presently call to_string() on a TransactionOutput that would just be unused code.

so, given the context of `TransactionOutput` private usage and above rationale, please let me know what changes are needed if any.   Personally I'm content to leave as-is, since I think we have bigger fish to fry.

Reply:
> I think we have bigger fish to fry.

Agreed. Given the context, no change is necessary. I would suggest adding a comment summarizing your answer, just so that we avoid the temptation to fix what seems on the surface like an obvious todo. Also, maybe motivate why colon delimitation is used. (I didn't realize until now but I guess that's because of clap?)

Reply:
I think I was just trying to come up with an easy (human) way to pass multiple outputs on the command line.   The send-to-many usage presently looks like:

```
Usage: neptune-cli send-to-many <OUTPUTS>... <FEE>

Arguments:
  <OUTPUTS>...  format: address:amount address:amount ...
  <FEE>
```

Which I think you will agree is pretty obvious/intuitive.  Outputs are just space and colon delimited.

We could think of more complicated schemes that clap could parse directly, such as --addr1=a, amount1=y --addr2=a amount2=z.  Or passing in json, or any number of things.

I found the space/colon delimiting to be the most intuitive for human usage, and not too bad for parsing.  But I'm open to suggestions should we wish to change it.

I'm definitely aware that send-to-many will be unwieldy to pass as CLI args no matter what we do.  This is because generation addresses are so lengthy.